lib/base_service.js

```diff
@@ -157,9 +157,17 @@ BaseService.prototype._sanitizeConfig = function(conf, options) {
         // the number of workers has been supplied
         // on the command line, so honour that
         conf.num_workers = options.num_workers;
-    } else if (conf.num_workers === 'ncpu' || typeof conf.num_workers !== 'number') {
-        // use the number of CPUs
-        conf.num_workers = os.cpus().length;
+    }
+    if (typeof conf.num_workers !== 'number') {
+        if (/^(?:(?:ncpu)|[\s\(\)\+\-\*\/\.\d])+$/.test(conf.num_workers)) {
```

Comment:
`(?:ncpu[\s\)\+\-\*\/])` would be more precise, so we don't have stuff like `ncpu(`, `ncpu.2`, etc